integration/test/Test/LegalHold.hs

```diff
@@ -274,3 +280,522 @@ testLHDeleteClientManually = do
+    approveLegalHoldDevice tid (bob %. "qualified_id") defPassword
+      >>= assertLabel 412 "legalhold-not-pending"
+
+    withMockServer lhMockApp \lhPort chan -> do
+      legalholdWhitelistTeam tid alice
+        >>= assertStatus 200
+      postLegalHoldSettings tid alice (mkLegalHoldSettings lhPort)
+        >>= assertStatus 201
+      requestLegalHoldDevice tid alice bob
+        >>= assertStatus 201
+
+      let uidsAndTidMatch val = do
+            actualTid <-
+              lookupFieldM val "team_id"
+                >>= lift . asString
+            actualUid <-
+              lookupFieldM val "user_id"
+                >>= lift . asString
+            bobUid <- lift $ objId bob
+
+            -- we pass the check on equality
+            unless ((actualTid, actualUid) == (tid, bobUid)) do
+              mzero
+
+      checkChanVal chan uidsAndTidMatch
+
+      -- the team owner cannot approve for bob
+      approveLegalHoldDevice' tid alice bob defPassword
+        >>= assertLabel 403 "access-denied"
+      -- bob needs to provide a password
+      approveLegalHoldDevice tid bob "wrong-password"
+        >>= assertLabel 403 "access-denied"
+      -- now bob finally found his password
+      approveLegalHoldDevice tid bob defPassword
+        >>= assertStatus 200
+
+      let matchAuthToken val =
+            lookupFieldM val "refresh_token"
+              >>= lift . asString
+
+      checkChanVal chan matchAuthToken
+        >>= renewToken bob
+        >>= assertStatus 200
+
+      bobId <- objId bob
+      lhdId <-
+        BrigI.getClientsFull bob [bobId] `bindResponse` \resp -> do
+          [lhd] <-
+            resp.json %. bobId
+              & asList
+              >>= filterM \val -> (== "legalhold") <$> (val %. "type" & asString)
+          lhd %. "id" & asString
+
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "client.id" `shouldMatch` lhdId
+        resp.json %. "status" `shouldMatch` "enabled"
+
+      replicateM 2 do
+        objId $ addClient bob def `bindResponse` getJSON 201
+        >>= traverse_ \client ->
+          awaitNotification bob client noValue isUserClientAddNotif >>= \notif -> do
+            notif %. "payload.0.client.type" `shouldMatch` "legalhold"
+            notif %. "payload.0.client.class" `shouldMatch` "legalhold"
+
+      -- the other team members receive a notification about the
+      -- legalhold device being approved in their team
+      for_ [alice, charlie] \user -> do
+        client <- objId $ addClient user def `bindResponse` getJSON 201
+        awaitNotification user client noValue isUserLegalholdEnabledNotif >>= \notif -> do
+          notif %. "payload.0.id" `shouldMatch` objId bob
+      for_ [ollie, sandy] \outsider -> do
+        outsiderClient <- objId $ addClient outsider def `bindResponse` getJSON 201
+        assertNoNotifications outsider outsiderClient Nothing isUserLegalholdEnabledNotif
```

Comment:
i don't understand.  how can the clients that are added after event X has been sent to everybody still see the event?

Reply:
I think I’ll have to ask you what you mean by this question?